src/ucp/core/ucp_ep.c

```diff
@@ -1143,10 +1144,26 @@ static void ucp_ep_set_lanes_failed(ucp_ep_h ep, uct_ep_h *uct_eps)
     }
 }
 
+static int ucp_should_uct_ep_unprogress(uct_ep_h uct_ep)
```

Comment:
need to add `context->config.ext.adaptive_progress` and `!ucp_ep_has_cm_lane` conditions, so I propose to change semantic to
`static int ucp_ep_can_unprogress_lane(ucp_ep_h ep, ucp_lane_index_t lane)`

Reply:
done. but i still use `uct_ep_h uct_ep` instead of `ucp_lane_index_t lane` because `ucp_ep_set_lanes_failed()` has been applied on `ep->uct_eps` before calling `ucp_ep_can_unprogress_lane()` we can not get `uct_ep` via `ep->uct_eps[lane]`.

Reply:
can we `ucp_ep_set_lanes_failed()` before `ucp_ep_set_lanes_failed()`?